src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +70,19 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+
+    /**
+     *
+     * @return true if the command mutates data else it will return false
+     */
+    public boolean isMutating() {
+        if(this.getClass().equals(ClearCommand.class) || this.getClass().equals(DeleteCommand.class) ||
+                this.getClass().equals(AddCommand.class)){
+            return true;
```

Comment:
as discussed during the tutorial, it is not recommended to include default behavior.

> if(this.getClass().equals(ClearCommand.class) || this.getClass().equals(DeleteCommand.class) ||
                this.getClass().equals(AddCommand.class)){

Also it is not good to do something like this as it defeats the purpose of polymorphism